File: src/BurstChat.Signal/Models/Payload.cs
namespace BurstChat.Signal.Models;

public class Payload<T>
    where T : notnull
{
    public string SignalGroup { get; set; }

    public T Content { get; set; }

    public Payload(string signalGroup, T content)
    {
        SignalGroup = signalGroup;
        Content = content;
    }
}

File: src/BurstChat.Client/src/app/models/signal/payload.ts
/**
 * This interface is the object sent from the BurstChat signal server that contains information
 * about the signalGroup and its content.
 * * @export
 * @interface Payload
 * @template T The type contained within the payload.
 */
export interface Payload<T> {

    signalGroup: string;
    content: T;

}
